repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala

```diff
@@ -78,16 +79,21 @@ object Main extends Logging {
     }
     sparkContext = new SparkContext(conf)
     logInfo("Created spark context..")
+    sqlContextWithHive = conf.getBoolean("spark.sql.hive.context", true)
     sparkContext
   }
 
   def createSQLContext(): SQLContext = {
-    val name = "org.apache.spark.sql.hive.HiveContext"
+    val name = sqlContextWithHive match {
+      case true => "org.apache.spark.sql.hive.HiveContext"
+      case false => "org.apache.spark.sql.SQLContext"
+    }
+
     val loader = Utils.getContextOrSparkClassLoader
     try {
       sqlContext = loader.loadClass(name).getConstructor(classOf[SparkContext])
         .newInstance(sparkContext).asInstanceOf[SQLContext]
-      logInfo("Created sql context (with Hive support)..")
+      logInfo("Created sql context with " + name)
     } catch {
       case _: java.lang.ClassNotFoundException | _: java.lang.NoClassDefFoundError =>
         sqlContext = new SQLContext(sparkContext)
```

Comment:
I don't know how to fix this, but the logic seems to be weird if SQLContext is used, then in the exception handler the context is created again.